clients/src/main/java/org/apache/kafka/clients/admin/Admin.java

```diff
@@ -1823,6 +1824,41 @@ default ListShareGroupsResult listShareGroups() {
         return listShareGroups(new ListShareGroupsOptions());
     }
 
+
+    /**
+     * Add the provided application metric for subscription.
+     * This metric will be added to this client's metrics
+     * that are available for subscription and sent as
+     * telemetry data to the broker.
+     * The provided metric must map to an OTLP metric data point
+     * type in the OpenTelemetry v1 metrics protobuf message types.
+     * Specifically, the metric should be one of the following:
+     * <ul>
+     *  <li>
+     *     `Sum`: Monotonic total count meter (Counter). Suitable for metrics like total number of X, e.g., total bytes sent.
+     *  </li>
+     *  <li>
+     *     `Gauge`: Non-monotonic current value meter (UpDownCounter). Suitable for metrics like current value of Y, e.g., current queue count.
+     *  </li>
+     * </ul>
+     * Metrics not matching these types are silently ignored.
+     * Executing this method for a previously registered metric is a benign operation and results in updating that metrics entry.
+     *
+     * @param metric The application metric to register
+     */
+    void registerMetricForSubscription(KafkaMetric metric);
```

Comment:
Yes, but @lucasbru's comment was related to using a Collection<Kafka Metric> to move to Collection<? extends Metric> but since the KIP changed to accept single metrics, it makes sense to use `KafkaMetric`.  That approach is consistent with the [Metrics.registerMetric](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java#L589) and the [MetricsReporter.metricsChange](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/metrics/MetricsReporter.java#L45) methods.

Reply:
Seems you missed my first qustions:

> Are existing client metrics isolated from app metrics? Or could it happen that a app metric would overwrite an existing one? I could not find any guard?
>
> Similar for unregisterMetricFromSubscription below: could this delete a client metric? Could not find a guard either.
> 
> If both are not isolated, we might want to put a guard in place, and let each method throw an exception if an existing client metric would be messed with? (And update the JavaDocs accordingly)